### PR TITLE
add nyc and codecov for coverage reporting

### PR DESCRIPTION
**Pull request by BarryThePenguin**
**What kind of change is this? (Bug Fix, Feature...)**

Chore

**What is the current behavior (You can also link to an issue)**

Currently only `/lib` files are linted, and coverage isn't reported

**What is the new behavior this introduces (if any)**

Will lint all files (`/lib` and `/tests`), and upload coverage report to codecov.io

**Does this introduce any breaking changes?**

nope

**Does the PR fulfill these requirements?**
- [x] Tests for the changes have been added
- [x] Docs have been added or updated

**Other Comments**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,12 +26,13 @@
...
     "eslint": "^3.16",
-    "istanbul": "^0.4",
-    "mocha": "^3.2"
+    "mocha": "^3.2",
+    "nyc": "^10.3.2"
   },
   "scripts": {
-    "test": "istanbul cover node_modules/mocha/bin/_mocha -- test/*.js",
-    "lint": "node_modules/.bin/eslint lib/*.js lost.js"
+    "test": "nyc mocha && codecov",
```
Comment (BarryThePenguin) [inline package.json:35]: nyc is a cli for istanbul

mocha looks in `/test` by default

File: package.json [hunk 2]
```diff
@@ -26,12 +26,13 @@
...
+    "lint": "eslint --ignore-path .gitignore ."
```
Comment (BarryThePenguin) [inline package.json:36]: make eslint lint all files, while ignore files included in the `. gitignore`

File: test/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    'env': {
+        'mocha': true,
```
Comment (BarryThePenguin) [inline test/.eslintrc.js:3]: set the mocha environment up for tests, adds globals like `describe`, `it`, etc

**Comment by peterramsing** [comment]
@BarryThePenguin, as always you're awesome! I'm gonna tackle some of the other MR's first and then get back to this one as this one is more of a chore.

**Pull request merged**